### PR TITLE
test: switch tab panel tests to react testing library

### PR DESCRIPTION
**Pull request by lsprr**
Issue #1159

This PR switches the Tab Panel tests to React Testing Library.

### Validation steps

- Run yarn test
- See that there are coverage percentages in the table
- Verify that the tests are converted properly

### CODE REVIEW

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React, { useRef } from 'react';
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:1]: Very minor nit on the test file name, it should be `TabPanel.test.tsx`

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        <TabPanel>
+          <p>a simple paragraph</p>
+          <p>a complicated paragraph</p>
+        </TabPanel>
+      </div>
+    );
+  };
+
+  render(<TabPanelwithRef />);
+  await waitFor(() => {
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:19]: Just curious, what was the `waitFor` added for? This seems to work without it.

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    expect(screen.getAllByRole('tabpanel')).toHaveLength(1);
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:20]:
```suggestion
    expect(screen.getAllByRole('tabpanel')).toBeInTheDocument();
```

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React, { useRef } from 'react';
+import { render, screen, waitFor } from '@testing-library/react';
+import { TabPanel } from './';
+
+test('should render tab panel with two paragraphs correctly', async () => {
+  const TabPanelwithRef = () => {
+    const ref = useRef(null);
+    return (
+      <div ref={ref}>
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:9]: To be perfectly honest, I don't really remember why we have this wrapped in a ref, so we may be able to omit the wrapping `div` and just render `TabPanel` directly for this and the other examples.

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+test('should pass classNames through', async () => {
+  const TabPanelwithRef = () => {
+    const ref = useRef(null);
+    return <TabPanel ref={ref} className="find--me" />;
+  };
+
+  render(<TabPanelwithRef />);
+  await waitFor(() => {
+    expect(screen.getByRole('tabpanel')).toHaveClass('find--me');
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:32]:
```suggestion
    expect(screen.getByRole('tabpanel')).toHaveClass('TabPanel', 'find--me');
```

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+  };
+
+  render(<TabPanelwithRef />);
+  await waitFor(() => {
+    expect(screen.getByRole('tabpanel')).toHaveAttribute(
+      'id',
+      'I am a panelId'
+    );
+  });
+});
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:49]: This is an older test so it's missing a few newer tests that we started to require, including testing that `ref` prop is supported, and running the component through axe.

**Review by scurker — CHANGES_REQUESTED** [review]

File: packages/react/src/components/Tabs/tabPanel.test.tsx (hunk 2 above)
Comment (lsprr) [reply packages/react/src/components/Tabs/tabPanel.test.tsx:19]: Thank you for pointing that out. I forgot to remove it, and it is not necessary to include.

File: packages/react/src/components/Tabs/tabPanel.test.tsx [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+  const ref = React.createRef<HTMLDivElement>();
+  const TabPanelwithRef = () => {
+    return <TabPanel ref={ref}>Content</TabPanel>;
+  };
+
+  render(<TabPanelwithRef />);
+  await waitFor(() => {
+    expect(ref.current).toBeInTheDocument();
+    expect(ref.current).toHaveTextContent('Content');
+  });
```
Comment (scurker) [inline packages/react/src/components/Tabs/tabPanel.test.tsx:36]: Just a very very minor nit, I think we can just render the tab panel with the ref directly.

```tsx
render(<TabPanel ref={ref}>Content</TabPanel>)
```

And I think based on the above discussion we can also omit the `waitFor`.

**Review by scurker — CHANGES_REQUESTED** [review]
One last change, and we should be good to go.

**Review by scurker — APPROVED** [review]

**Comment by scurker** [comment]
🚨 Reviewed for Security 🚨

**Pull request merged**